cloudferrylib/os/compute/nova_compute.py

```diff
@@ -57,26 +58,29 @@ def get_client(self, params=None):
                                   params.cloud.tenant,
                                   params.cloud.auth_url)
 
-    def _read_info_quotas(self, info):
-        user_quotas_cmd = ("SELECT user_id, project_id, resource, "
-                           "hard_limit FROM project_user_quotas WHERE "
-                           "deleted = 0")
-        for quota in self.mysql_connector.execute(user_quotas_cmd):
-            info['user_quotas'].append(
-                {'quota': {'user_id': quota[0],
-                           'project_id': quota[1],
-                           'resource': quota[2],
-                           'hard_limit': quota[3]},
-                 'meta': {}})
-
-        project_quotas_cmd = ("SELECT project_id, resource, hard_limit "
-                              "FROM quotas WHERE deleted = 0")
-        for quota in self.mysql_connector.execute(project_quotas_cmd):
-            info['project_quotas'].append(
-                {'quota': {'project_id': quota[0],
-                           'resource': quota[1],
-                           'hard_limit': quota[2]},
-                 'meta': {}})
+    def _read_info_quotas(self):
+        service_tenant_id = \
+            self.identity.get_tenant_id_by_name(self.config.cloud.service_tenant)
+        tenant_ids = [tenant.id for tenant in self.identity.get_tenants_list()
+                      if tenant.id != service_tenant_id]
+        user_ids = [user.id for user in self.identity.get_users_list()
+                    if not self.identity.roles_for_user(user.id, service_tenant_id)]
```

Comment:
Retrieving the list of all users for real production environment may take hours if keystone uses corporate LDAP as a backend. Let's avoid making this call.